pulumi/folio/index.ts

```diff
@@ -0,0 +1,203 @@
+        transformations: [(args: any) => {
+            if (args.type === "aws:ec2/vpc:Vpc" || args.type === "aws:ec2/subnet:Subnet") {
+                return {
+                    props: args.props,
+                    opts: pulumi.mergeOptions(args.opts, { ignoreChanges: ["tags"] }),
+                };
+            }
+            return undefined;
+        }],
+    }
+);
+
+// Create the security group. We need a custom security group since we're going
+// to expose non-standard ports for things like edge modules. This is using the pulumi
+// Classic API. Although Crosswalk (awsx) has its own SecurityGroup type, Cluster
+// wants a Classic SecurityGroup type. See Cluster.clusterSecurityGroup below
+// where we pass this into the cluster.
+const sgName = "folio-sg";
+const sg = new aws.ec2.SecurityGroup(sgName, {
+    tags: { "Name": sgName, ...tags },
+    description: "Security group for FOLIO EKS cluster.",
+    vpcId: vpc.id,
+    ingress: [{
+        description: "Allow inbound traffic on 443",
+        fromPort: 443,
+        toPort: 443,
+        protocol: "tcp",
+        cidrBlocks: ["0.0.0.0/0"]
+    }, {
+        description: "Allow inbound traffic on 9000 for edgeconexion",
+        fromPort: 9000,
+        toPort: 9000,
+        protocol: "tcp",
+        cidrBlocks: ["0.0.0.0/0"]
+    }],
+    egress: [{
+        description: "Allow outbound traffic",
+        fromPort: 0,
+        toPort: 0,
+        protocol: "-1",
+        cidrBlocks: ["0.0.0.0/0"]
+    }]
+});
+
+// Create our own IAM role and profile which we can bind nto the cluster's
+// NodeGroup. Cluster will also create a default for us, but we show here how
+// to create and bind our own.
+const managedPolicyArns: string[] = [
+    "arn:aws:iam::aws:policy/AmazonEKSWorkerNodePolicy",
+    "arn:aws:iam::aws:policy/AmazonEKS_CNI_Policy",
+    "arn:aws:iam::aws:policy/AmazonEC2ContainerRegistryReadOnly",
+];
+
+// Creates a role and attaches the EKS worker node IAM managed policies.
+export function createRole(name: string): aws.iam.Role {
+    const role = new aws.iam.Role(name, {
+        tags: { "Name": name, ...tags },
+
+        assumeRolePolicy: aws.iam.assumeRolePolicyForPrincipal({
+            Service: "ec2.amazonaws.com",
+        }),
+    });
+
+    let counter = 0;
+    for (const policy of managedPolicyArns) {
+        // Create RolePolicyAttachment without returning it.
+        const rpa = new aws.iam.RolePolicyAttachment(`${name}-policy-${counter++}`,
+            { policyArn: policy, role: role },
+        );
+    }
+
+    return role;
+}
+
+// We could create multiple roles and instance profiles here if we wanted to
+// perhaps to apply different profiles to different node groups.
+const role = createRole("folio-worker-role");
+const instanceProfile = new aws.iam.InstanceProfile("folio-instance-profile", { role: role });
+
+// Create an EKS cluster.
+const clusterName = "folio-cluster";
+const cluster = new eks.Cluster(clusterName, {
```

Comment:
Pin the kubernetes version https://www.pulumi.com/registry/packages/eks/api-docs/cluster/#version_nodejs